updater.go

```diff
@@ -121,14 +137,37 @@ func Manage(conf *Config) (*Updater, error) {
 		}
 
 		if updater.conf.Schedule.Interval != 0 {
-			for {
-				time.Sleep(updater.conf.Schedule.Interval)
-				logInfo("Scheduled upgrade check after %s.\n", updater.conf.Schedule.Interval)
-				err := updater.CheckNow()
-				if err != nil {
-					logError("Upgrade error: %v\n", err)
+			go func() {
+				for {
+					time.Sleep(updater.conf.Schedule.Interval)
+					logInfo("Scheduled upgrade check after %s.\n", updater.conf.Schedule.Interval)
+					err := updater.CheckNow()
+					if err != nil {
+						logError("Upgrade error: %v\n", err)
+					}
 				}
-			}
+			}()
+		}
+
+		if updater.conf.Schedule.At.Repeating != None {
+			go func() {
+				for {
+					now := time.Now()
+					var next time.Time
+					switch updater.conf.Schedule.At.Repeating {
+					case Minutly:
```

Comment:
This really should be factored out into another method that determines what the "next time" for a schedule.
That can also then be tested - this is some pretty clever date manipulation that has no tests.